web/src/components/FormActivitySchedule.js

```diff
@@ -0,0 +1,176 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Box, Button } from 'rebass';
+import { Field, FieldArray, reduxForm } from 'redux-form';
+
+import { Input } from './Inputs';
+import SectionHeader from './SectionHeader';
+
+const AddButton = ({ fields, meta: { error, submitFailed } }) => (
+  <Box>
+    <Button bg="black" onClick={() => fields.push({})}>
+      Add another milestone
+    </Button>
+    {submitFailed && error && <div>{error}</div>}
+  </Box>
+);
+
+AddButton.propTypes = {
+  fields: PropTypes.object.isRequired,
+  meta: PropTypes.object.isRequired
+};
+
+const Milestones = ({ fields }) =>
+  fields.map((milestone, idx) => (
+    <tr>
+      <td>{idx + 1}</td>
+      <td>
+        <Field
+          name={`${milestone}.name`}
+          component={Input}
+          label="name"
+          hideLabel
+        />
+      </td>
+      <td>
+        <Field
+          name={`${milestone}.status`}
+          component={Input}
+          label="status"
+          hideLabel
+        />
+      </td>
+      <td>
+        <Field
+          name={`${milestone}.plannedStart`}
+          component={Input}
+          label="planned start date"
+          hideLabel
+        />
+      </td>
+      <td>
+        <Field
+          name={`${milestone}.actualStart`}
+          component={Input}
+          label="actual start date"
+          hideLabel
+        />
+      </td>
+      <td>
+        <Field
+          name={`${milestone}.plannedEnd`}
+          component={Input}
+          label="planned end date"
+          hideLabel
+        />
+      </td>
+      <td>
+        <Field
+          name={`${milestone}.actualEnd`}
+          component={Input}
+          label="actual end date"
+          hideLabel
+        />
+      </td>
+      <td>
+        <button
+          type="button"
+          title="Remove Goal"
+          onClick={() => fields.remove(idx)}
+        >
+          Remove milestone
+        </button>
+      </td>
+    </tr>
+  ));
+
+Milestones.propTypes = {
+  fields: PropTypes.object.isRequired
+};
+
+const FormActivitySchedule = ({
+  handleSubmit,
+  pristine,
+  reset,
+  submitting
+}) => (
+  <form onSubmit={handleSubmit}>
+    <SectionHeader>
+      List the major milestones you&apos;re working towards as part of this
+      activity
+    </SectionHeader>
+
+    <table>
+      <thead>
+        <tr>
+          <th />
+          <th>Milestone</th>
+          <th>Status</th>
+          <th>Planned start date</th>
+          <th>Actual start date</th>
+          <th>Planned end date</th>
+          <th>Actual end date</th>
+          <th />
+        </tr>
+      </thead>
+      <tbody>
+        <FieldArray name="milestones" component={Milestones} />
+      </tbody>
+    </table>
+
+    <FieldArray name="milestones" component={AddButton} />
```

Comment:
Not sure I understand this bit -- why is `AddButton` being used as a FieldArray? Is that just so it gets the `fields` prop for `milestones`?

Reply:
Yeah, that's exactly why.  It was the most obvious way of getting access to the `fields` prop - and faster than digging through redux-form docs.  😛

Reply:
Ok, i don't love that, but we can figure out a cleaner way and refactor soon

Reply:
I also don't love it.  I think this falls under the conversation we were having about how to take the `redux-form` state and merge it into our "real" state.